packages/sit-onyx/src/utils/comparator.ts

```diff
@@ -0,0 +1,13 @@
+type FlatObject = Record<string, string | boolean | number | null | undefined>;
+
+/**
+ * Compares two flat objects (whose values are only primitives, no arrays or objects)
+ * @returns whether all values match, ignoring in which order they appear
+ */
+export const areObjectsFlatEqual = (obj1: FlatObject, obj2: FlatObject): boolean => {
+  if (Object.keys(obj1).length !== Object.keys(obj2).length) return false;
```

Comment:
I am not sure if this check is correct, example:
`{ a: "foo", b: undefined }` and `{ a: "foo" }` are equal but have a different number of keys

Can you add a test case for this?